lib/meta-attachments.php

```diff
@@ -0,0 +1,186 @@
+	 *
+	 * @var string $meta_location
+	 */
+	private string $meta_location = 'side';
+
+	/**
+	 * The meta add attachment button label.
+	 *
+	 * @var string $attachment_button_label
+	 */
+	private string $attachment_button_label = '';
+
+	/**
+	 * Class constructor.
+	 *
+	 * @param array $args {
+	 *     An array of arguments.
+	 *
+	 *     @type string $meta_key Meta key to register (stored in post meta)
+	 *     @type array $post_types Array of post types to list this meta box on.
+	 *     @type string $meta_label Label for insert attachment button (see add_meta_box).
+	 *     @type string $meta_priority Priority of meta box (see add_meta_box).
+	 *     @type string $meta_location Location of meta box (see add_meta_box).
+	 * }
+	 */
+	public function __construct( $args ) {
+		// Set defaults.
+		$defaults = array(
+			'id' => 'course_maker_featured_article_image',
+			'meta_key' => 'course_maker_featured_article_image',
+			'post_types' => array(
+				'post',
+			),
+			'meta_label' => __( 'Add Image', 'coursemaker' ),
+			'meta_priority' => 'high',
+			'meta_location' => 'side',
+			'attachment_button_label' => __( 'Add Image', 'coursemaker' ),
+		);
+
+		// Merge arguments.
+		$args = wp_parse_args( $args, $defaults );
+
+		// Set class variables.
+		foreach ( $args as $key => $value ) {
+			$this->{ $key } = $value;
+		}
+	}
+
+	/**
+	 * Run the hooks associated with this class.
+	 */
+	public function run_hooks() {
+		foreach ( $this->post_types as $post_type ) {
+			add_action( 'add_meta_boxes_' . $post_type, array( $this, 'add_meta_box_callback' ) );
+		}
+
+		// Load the attachment script.
+		add_action( 'admin_enqueue_scripts', 'course_maker_display_featured_article_image_scripts' );
+	}
+
+	/**
+	 * Registers the Attachment Meta box.
+	 *
+	 * @param object $post The current post object.
+	 */
+	public function add_meta_box_callback( $post ) {
+		add_meta_box(
+			sanitize_title( $this->meta_slug ),
+			$this->meta_label,
+			array( $this, 'display_meta_box' ),
+			get_post_type( $post ),
+			$this->meta_location,
+			$this->meta_priority
+		);
+	}
+
+	/**
+	 * Displays the meta box.
+	 *
+	 * @param object $post The current post object.
+	 */
+	public function display_meta_box( $post ) {
+		// Create a new nonce.
+		wp_nonce_field( basename( __FILE__ ), sanitize_title( $slug ) . '_nonce' );
+
+		$attachment_image = get_post_meta( $post->ID, $this->id, true );
+
+		// Get the _featured_article current value.
+		// $current_featured_article_value = get_post_meta( $post->ID, '_featured_article', true );
+
+		// Show the fields.
+		?>
+		<div class='inside'>
+
+			<div>
+				<button id="<?php echo esc_html( $this->id ); ?>" type="button" class="button-secondary">
+					<?php echo esc_html( $this->attachment_button_label ); ?>
+				</button>
+			</div>
+
+		</div>
+		<?php
+	}
+}
+
+/**
+ * Load post admin scripts for the featured article image.
+ */
+function course_maker_display_featured_article_image_scripts( $hook ) {
```

Comment:
This hasn't been placed in the class yet. I need to complete this part.

Reply:
Just need to create a jQuery plugin so the script can handle multiple inputs (e.g., 2 meta boxes on the same post type).